### PR TITLE
Fix flt file read for large npart

### PR DESCRIPTION
**Pull request by nathanieltarshish**
For `npart > 1.6 x 10^7`, rounding error generated by type converting from integer to float32 causes `npart_read` to differ from the true number of floats in the `flt_file`. This PR seeks to remedy this problem. We let the user set `flt_npart`, an (integer) parameter to be specified in `data.flt`. If not given, `flt_npart` defaults to zero.  Iff  `flt_npart=0`, a conditional in `flt_init_varia.F` sets `npart_read` equal to the first entry of the `flt_file` (the prior way `npart_read` was specified ). In all other cases, `npart_read=flt_npart` and the `flt_file` should be read in without issue.

### CODE REVIEW

File: code_fix_floater/flt_init_varia.F [hunk 1]
```diff
@@ -119,12 +119,19 @@ SUBROUTINE FLT_INIT_VARIA ( myThid  )
...
       IF ( ioUnit.GT.0 .AND. mapIniPos2Index ) THEN
 C--   Found a global file
-        WRITE(msgBuf,'(A,2I4,A,1P2E15.8)')
-     &    ' bi,bj=', bi, bj, ' , npart,max_npart=', tmp(1), tmp(6)
+C-------Start Nathaniel Hack----------------------
+        IF (flt_npart.EQ.0 ) THEN
+          npart_read = NINT(tmp(1))
+        ELSE
+          npart_read = flt_npart
+        ENDIF
```
Comment (rabernat) [inline code_fix_floater/flt_init_varia.F:127]: I like this...if you leave out flt_npart from data.flt, it still tries get it from the file.

**Comment by rabernat** [comment]
Have you actually tried this?

**Comment by nathanieltarshish** [comment]
Yes, it worked on a global run in CM2p6 (which failed prior to this change).

 I should probably make a superficial change to allot more digits when writing `npart` to STDOUT. I'll do that tonight.

File: code_fix_floater/flt_init_varia.F [hunk 2]
```diff
@@ -119,12 +119,19 @@ SUBROUTINE FLT_INIT_VARIA ( myThid  )
...
+
+        WRITE(msgBuf,'(A,2I4,A,I4)')
```
Comment (nathanieltarshish) [inline code_fix_floater/flt_init_varia.F]: This is the line I should change from `I4` to `I9` or so.  So that it gets written to STDOUT correctly.

**Comment by nathanieltarshish** [comment]
Ready to merge. The relevant chunk of STDOUT is now 

```
(PID.TID 0000.0001) FLT_INIT_VARIA: reading Floats from: flt_CM2p6_global_32deg.bin
(PID.TID 0000.0001)  MDS_READVEC_LOC: open file: flt_CM2p6_global_32deg.bin
(PID.TID 0000.0001)  bi,bj=   0   0 ,hacked npart_read=  37171843
```

**Pull request merged**